### DISCUSSION TITLE
Allow TraceQL results to be visualized

### DISCUSSION THREAD

**Discussion by spockz** [grafana/grafana#70517] [Dashboards, Panels & Visualizations]
On our current in-house tracing solution which stores spans in Elastic we use Kibana to do all kinds of ad-hoc queries, especially during major incident resolution. We are now moving to OpenTelemetry with Tempo and therefore TraceQL and I like it a lot so far. My only problem is that I couldn't find a way to make graphs out of traceql query results. I can graph something per trace, but I cannot graph over spans in those traces. It seems the result is a nested table and it looks like I need some way to flatten the table but I don't know how.

How can I make graphs for all spans in a traceql query result?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). This is phrased as a question, and we'd like questions about grafana to go to our[ community forum](https://community.grafana.com/). Feel free to make an [issue](https://github.com/grafana/grafana/issues) if you cannot find a solution and would like to request the feature.
